src/test/java/org/jitsi/meet/test/DesktopSharingTest.java

```diff
@@ -167,20 +116,69 @@ public void testDesktopSharingInPresence()
         catch (TimeoutException e)
         {
             assertEquals(
-                expectedResult, remoteVideoType[0],
-                "Wrong video type, maybe desktop sharing didn't work");
+                    expectedResult, remoteVideoType[0],
+                    "Wrong video type, maybe desktop sharing didn't work");
         }
+    }
 
-        // allow the participant to leave
-        try
-        {
-            waitEndSignal.await(10, TimeUnit.SECONDS);
-            print("End DesktopSharingTest.");
-        }
-        catch (InterruptedException e)
-        {
-            e.printStackTrace();
-        }
-        ensureTwoParticipants();
+    /**
+     * Starts desktop sharing.
+     */
+    private void startDesktopSharing()
+    {
+        TestUtils.waitForElementByXPath(
+                getParticipant2().getDriver(),
+                DS_BUTTON_XPATH,
+                5);
+        MeetUIUtils.clickOnToolbarButton(getParticipant2().getDriver(),
+                DS_BUTTON_ID);
+        TestUtils.waitForElementContainsClassByXPath(
+                getParticipant2().getDriver(), DS_BUTTON_XPATH,
+                "toggled", 2);
+    }
+
+    /**
+     * Stops desktop sharing.
+     */
+    private void stopDesktopSharing()
+    {
+        TestUtils.waitForElementByXPath(
+                getParticipant2().getDriver(),
+                DS_BUTTON_XPATH,
+                5);
+        MeetUIUtils.clickOnToolbarButton(getParticipant2().getDriver(),
+                DS_BUTTON_ID);
+        TestUtils.waitForElementNotContainsClassByXPath(
+                getParticipant2().getDriver(), DS_BUTTON_XPATH,
+                "toggled", 2);
+    }
+
+    /**
+     * Checks the video layout on the other side, after we imitate
+     * desktop sharing.
+     * @param isScreenSharing <tt>true</tt> if SS is started and <tt>false</tt>
+     *                        otherwise.
+     */
+    private void checkExpandingDesktopSharingLargeVideo(boolean isScreenSharing)
+    {
+        // check layout
+        new VideoLayoutTest().driverVideoLayoutTest(
```

Comment:
I hate that we need to create test instances somewhere else, but it would be probably too much to refactor this in this PR...